### ISSUE TITLE
Tooltip: Malfunction inside Links

### ISSUE THREAD

**Issue by havist** [primefaces/primevue#5030]
### Describe the bug

There is a serious bug with Tooltip when it's used inside Links. When so, it causes the infinite recursion when trying to bind listeners using Tooltip's `bindScrollListener()` method resulting with `Uncaught RangeError: Maximum call stack size exceeded`. The problem lies in `DomHandler.getScrollableParents()` utility that cannot properly read parent nodes. That's because in described case the condition checking parent for **null** in DomHandler's code below does not work (parent is undefined instead).

```
getParents: function getParents(element) {
    var parents = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : [];
    var parent = this.getParentNode(element);
    return parent === null ? parents : this.getParents(parent, parents.concat([parent]));
},
```

### Reproducer

https://codesandbox.io/p/sandbox/hungry-gagarin-nm9xwy

### PrimeVue version

3.45.0

### Vue version

3.x

### Language

ALL

### Build / Runtime

Vue CLI App

### Steps to reproduce the behavior

Try the code in Sandbox and see the console.

### Expected behavior

Tooltip should work w/o errors.

**Comment by tugcekucukoglu** [comment]
Cannot open the codesandbox demo. Could you update the link?

**Comment by havist** [comment]
Well, could you try [this share link](https://codesandbox.io/p/devbox/hungry-gagarin-nm9xwy?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clqzambif0009356i6bq0yg6e%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clqzambie0002356ipims7aww%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clqzambif0006356icgnhnsw8%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clqzambif0008356iu107slww%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clqzambie0002356ipims7aww%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clqzambie0001356i3fnzl834%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clqzanygn002i356hei40k900%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A5%252C%2522startColumn%2522%253A1%252C%2522endLineNumber%2522%253A12%252C%2522endColumn%2522%253A1%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.vue%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clqzambie0002356ipims7aww%2522%252C%2522activeTabId%2522%253A%2522clqzanygn002i356hei40k900%2522%257D%252C%2522clqzambif0008356iu107slww%2522%253A%257B%2522id%2522%253A%2522clqzambif0008356iu107slww%2522%252C%2522activeTabId%2522%253A%2522clqzbi9u300s2356h4trgesp4%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522id%2522%253A%2522clqzbi9u300s2356h4trgesp4%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%257D%252C%2522clqzambif0006356icgnhnsw8%2522%253A%257B%2522id%2522%253A%2522clqzambif0006356icgnhnsw8%2522%252C%2522activeTabId%2522%253A%2522clqzbi68000pg356hgfe22uol%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522clqzbi68000pg356hgfe22uol%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D) then, please?

**Comment by vaibs30** [comment]
Do we have an update here?
It is certainly reproducible and in the case of Safari, the whole page hangs.

**Comment by mertsincan** [comment]
Thanks a lot, @vaibs30! We'll check before the next version release. Sorry for the delayed response!

Best Regards,

**Comment by mattlyons0** [comment]
Hey I noticed this was supposed to be released in 3.47.0 recently but I'm not seeing it as fixed nor the changes in the published sources.

Here's an updated sandbox link to `3.47.1` where you can still reproduce the issue: https://codesandbox.io/p/devbox/hopeful-roman-dwn2qp
And if I open the sources of the installed package (`node_modules/primevue/utils/utils.js`) I still see the old logic in the built source:
```
getParents: function getParents(element) {
        var parents = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : [];
        var parent = this.getParentNode(element);
        return parent === null ? parents : this.getParents(parent, parents.concat([parent]));
      },
```

**Comment by mertsincan** [comment]
Sorry for the confusion :/ There was confusion while merging from a different branch to master. I fixed it now. https://github.com/primefaces/primevue/issues/5146 We will release the new version today. Thanks a lot for your report!

**Comment by mattlyons0** [comment]
No worries! Can confirm I'm seeing the fix in `3.47.2`. Thanks
